server/src/engine/queue.ts

```diff
@@ -53,23 +52,52 @@ function jobToEnqueuedJob(job: jobs.Job) {
 }
 
 function emptyWaitingForPriorityQueue() {
-  const values = _.values(waitingForPriorityQueue)
+  const values = Object.values(waitingForPriorityQueue)
   waitingForPriorityQueue = {}
   return values
 }
 
 let enqueueDebounced: () => void = makeEnqueueDebounced()
 
 function handleEnqueue() {
-  if (_.isEmpty(waitingForPriorityQueue)) {
+  if (Object.keys(waitingForPriorityQueue).length === 0) {
     return
   }
   priorityQueue.push(...emptyWaitingForPriorityQueue())
   startQueue()
 }
 
 function makeEnqueueDebounced() {
```

Comment:
Perhaps this debouce can be in a separate PR?

Reply:
Then we'd have to keep in the dependency and make a third PR

Reply:
I think breaking this PR down to a few smaller PRs reduces the risk.

Reply:
Yep, exactly.